src/app/static/src/app/store/errors/actions.ts

```diff
@@ -0,0 +1,64 @@
+import {ActionContext, ActionTree} from "vuex";
+import {RootState} from "../../root";
+import {ErrorsMutation} from "./mutations";
+import {ErrorsState} from "./errors";
+import {api} from "../../apiService";
+import {VersionInfo} from "../../generated";
+import {currentHintVersion} from "../../hintVersion";
+
+
+export interface ErrorsActions {
+    generateErrorReport: (store: ActionContext<ErrorsState, RootState>,
+                          payload: ErrorReportManualDetails) => void;
+}
+
+export interface ErrorReportManualDetails {
+    section: string,
+    description: string,
+    stepsToReproduce: string,
+    email: string
+}
+
+export interface ErrorReport extends ErrorReportManualDetails {
+    country: string,
+    projectName: string | undefined,
+    browserAgent: string,
+    timeStamp: string,
+    jobId: string,
+    versions: VersionInfo,
+    errors: Error[]
+}
+
+export const actions: ActionTree<ErrorsState, RootState> & ErrorsActions = {
+
+    async generateErrorReport(context, payload) {
+        const {dispatch, rootState, getters} = context
+        const data = {
+            email: payload.email || rootState.currentUser,
+            country: rootState.baseline.country || "no associated country",
+            projectName: rootState.projects.currentProject?.name || "no associated project",
```

Comment:
Ah, I'm afraid this needs to remain in root actions, because it won't be re-usable - it references the projects module, which won't exist in the new app.

Reply:
Right, so we'll need a separate action to generate the report for data exploration mode. I guess the alternative would be some ugly casting to any to check if state is actually a root state or not (has projects). But if we're going to have a separate dialog component for DE error reporting anyway, cleaner to have that as separate action.